lighthouse-plugin-publisher-ads/audits/ad-request-critical-path.js

```diff
@@ -96,6 +101,42 @@ function areSimilarRequests(r1, r2) {
   return r1.abbreviatedUrl == r2.abbreviatedUrl;
 }
 
+/**
+ * Computes self time for each request
+ * @param {SimpleRequest[]} requests A pre-sorted list of requests by start
+ *   time.
+ */
+function computeSelfTime(requests) {
+  let bottlneckRequest = requests[0];
+  bottlneckRequest.selfTime = bottlneckRequest.duration;
+
+  let scanEnd = bottlneckRequest.startTime;
+  for (let i = 1; i < requests.length - 1; i++) {
+    const current = requests[i];
+    if (current.endTime < scanEnd) {
+      // Overlaps with previous requests, skip to avoid double counting.
+      continue;
+    }
+    const left = Math.max(scanEnd, current.startTime);
+    const right = Math.min(bottlneckRequest.endTime, current.endTime);
+    if (left < right) {
+      // @ts-ignore selfTime is initialized elsewhere, so it won't be undefined.
+      bottlneckRequest.selfTime -= (right - left);
+    }
+    scanEnd = Math.max(scanEnd, right);
+    if (current.endTime > bottlneckRequest.endTime) {
+      // The next request is a potential bottleneck.
+      current.selfTime = current.endTime - left;
+      bottlneckRequest = current;
+    }
+  }
+  for (const request of requests) {
+    if (request.selfTime == null || request.selfTime < 100) {
```

Comment:
I think it's unclear that no self time means that the self time is short. I would prefer to either just keep the time, or use the string `< 100ms`

Reply:
Obsolete